### DISCUSSION TITLE
Client side hydration not working in production mode

### DISCUSSION THREAD

**Discussion by stevenamoody** [vercel/next.js#16053] [Help] [answered]
Hello!

I'm working through a project to upgrade an existing code base to the latest version of NextJS (9.5.1). I'm now running into an issue where client side hydration is not working correctly. We use a custom server implementation and most of our pages are SSR'd. When running in production mode with a page that is SSR'd
* getServerSideProps is correctly called and the page is partially rendered with the correct information
* Event listeners are not being attached to buttons. Clicking them does nothing.
* we have an async componentDidMount method on our _app.ts wrapper that doesn't get called.
* We have a carousel component that should be rendering a `ul` of `li` for each image fetched in getServerSideProps. No `li` elements are rendered to the dom

When i turn on dev mode, all these problems disappear and the page works correctly. I'm very confused how dev mode works differently and why these issues only appear when i turn it off. Any insight would be greatly appreciated.

**Comment by jamesmosier** [comment, marked as answer]
Do you happen to have a repository that reproduces this? That'd be super helpful in tracking down what might be causing this.

**Reply by stevenamoody** [reply]
We figured it out, we had a weird webpack config in our next config that was changing how chunking works. I removed it completely and it works now.

**Reply by jespergra** [reply]
I'm experiencing the same thing. I am using next-images and styled-components and using withImages() and withPWA() in the next.config.js.

Do you use that as well?

Console from Production.
![image](https://user-images.githubusercontent.com/6495801/98308599-be350c80-1fc8-11eb-87fa-f6f561c8c740.png)

Does it looks similar?

**Reply by MohamedJakkariya** [reply]
@steven-tey Could you please share your webpack config?

I've also faced some unexpected hydration issues in production mode.
